packages/strapi-plugin-graphql/services/Mutation.js

```diff
@@ -195,27 +195,39 @@ module.exports = {
 
       // Resolver can be a function. Be also a native resolver or a controller's action.
       if (_.isFunction(resolver)) {
+        const normalizedName = _.toLower(name);
+
+        let primaryKey;
+
+        if (plugin) {
+          primaryKey = strapi.plugins[plugin].models[normalizedName].primaryKey;
+        } else {
+          primaryKey = strapi.models[normalizedName].primaryKey;
+        }
+
         context.params = Query.convertToParams(
           options.input.where || {},
-          (plugin ? strapi.plugins[plugin].models[name] : strapi.models[name])
-            .primaryKey
+          primaryKey
         );
+
         context.request.body = options.input.data || {};
 
         if (isController) {
           const values = await resolver.call(null, context);
 
           if (ctx.body) {
             return {
-              [pluralize.singular(name)]: ctx.body,
+              [pluralize.singular(normalizedName)]: ctx.body,
             };
           }
 
           const body = values && values.toJSON ? values.toJSON() : values;
 
-          return {
-            [pluralize.singular(name)]: body,
-          };
+          return action
```

Comment:
action will always be set here I think ? this might be a problem for you no ?

Reply:
You're right, missed that. I don't think it will be an issue.